extensions/api/v1alpha1/sandboxwarmpool_types.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2025 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	sandboxv1alpha1 "sigs.k8s.io/agent-sandbox/api/v1alpha1"
+)
+
+// NOTE: json tags are required. Any new fields you add must have json tags for the fields to be serialized.
+// Important: Run "make" to regenerate code after modifying this file
+
+// SandboxWarmPoolSpec defines the desired state of SandboxWarmPool
+type SandboxWarmPoolSpec struct {
+	// Replicas is the desired number of sandboxes in the pool.
+	// This field is controlled by an HPA if specified.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Minimum=0
+	Replicas int32 `json:"replicas"`
+
+	// PodTemplate describes the pod spec that will be used to create sandboxes in the warm pool.
+	// +kubebuilder:validation:Required
+	PodTemplate sandboxv1alpha1.PodTemplate `json:"podTemplate"`
+}
+
+// SandboxWarmPoolStatus defines the observed state of SandboxWarmPool
+type SandboxWarmPoolStatus struct {
+	// Replicas is the total number of sandboxes in the pool.
+	// +optional
+	Replicas int32 `json:"replicas,omitempty"`
+
+	// ReadyReplicas is the number of sandboxes that are ready.
+	// +optional
+	ReadyReplicas int32 `json:"readyReplicas,omitempty"`
+
+	// Conditions represent the latest available observations of the warm pool's state.
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
```

Comment:
I don't see this used in the controller code

Reply:
To maintain a clean design, we should only include fields that are actively in use. Therefore, I recommend removing the `Conditions` field for the time being.

Regarding `Status.ReadyReplicas`, its usefulness to end-users is debatable. If the controller is responsible for assigning pods when they are claimed, and there is ambiguity around which pods are assigned and whether or not they will be ready, this field may not provide actionable information.